stacks-signer/src/v0/signer_state.rs

```diff
@@ -1201,6 +1216,26 @@ impl LocalStateMachine {
                 ))
             .cloned()
             .collect::<Vec<_>>();
-        Ok(Some(forked_txs))
+        forked_txs
+    }
+
+    /// If it has been 2 burn blocks since the origin of our replay set, and
+    /// we haven't produced any replay blocks since then, we should reset our replay set
+    ///
+    /// Returns a `bool` indicating whether the replay set should be reset.
+    fn handle_possible_replay_failsafe(
+        replay_state: &ReplayState,
+        new_burn_block: &NewBurnBlock,
+        reset_replay_set_after_fork_blocks: u64,
+    ) -> Result<bool, SignerChainstateError> {
+        let ReplayState::InProgress(_, replay_scope) = replay_state else {
+            // Not in replay - skip
+            return Ok(false);
+        };
+
+        let failsafe_height =
+            replay_scope.past_tip.burn_block_height + reset_replay_set_after_fork_blocks;
```

Comment:
If I'm understanding the scope construction correctly, I think the behavior here is roughly: if a replay set takes longer than 2 burn blocks to resolve, clear the replay set.

I think that behavior is actually fine. But I think the alternative we discussed was something like "if its been more than 2 burn blocks since a transaction in the replay set has been processed" which is a more restrictive condition (i.e., less likely to trigger). I'm okay with the less restrictive condition, but we should make sure to communicate it.